README Updates

ritz303 opened the pull request:
Updated the formatting of the README.md content so it renders properly on the Ansible Automation Hub site, and on this GitHub project page. Along with those changes, I made edits for grammar and our tech writing standards.

Fixes: Jira [SECURESIGN-2062](https://issues.redhat.com/browse/SECURESIGN-2062)

bkabrda commented on README.md:
```diff
@@ -2,10 +2,10 @@
 The purpose of this Ansible collection is to automate the deployment of the Red Hat Trusted Artifact Signer (RHTAS) service on Red Hat Enterprise Linux (RHEL).
 
-> [!IMPORTANT]
+**IMPORTANT:**
```
I think we should remove this notice altogether now, as we won't be shipping this as tech preview in 1.2.0.

bkabrda reviewed (changes requested):
Hi Aron, thanks a lot for this PR, it looks awesome. I left one comment inline and have two more general comments:

* The .gitignore file shouldn't be removed
* At one place in the README that is not touched by your PR, the "Requirements" section, there are two nested lists that aren't formatted correctly, because right now the nested lists start with 2 empty spaces while AAH apparently requires them to start with 4 empty spaces. Can you please fix these?

Once the points are addressed, I can approve and merge.

ritz303 commented:
@bkabrda : Thank you for taking time to review my updates!  Sorry for removing the `.gitignore` file by accident.  I removed the Tech Preview content, and fixed the indenting for the bulleted list under the Requirements section (good catch!).

bkabrda reviewed (approved):
LGTM, thanks! Feel free to merge
